src/genomic_features/_core/filters.py

```diff
@@ -115,6 +115,56 @@ def convert(self) -> ibis.expr.deferred.Deferred:
             return ibis.deferred[list(self.columns())[0]].isin(self.value)
 
 
+#  ------------------------ OVERLAP TYPE: any ------------------------ #
+# Range:       |==============================|
+# Annotation: |---|     |-------|           |-----| |---|
+# Selected:    ***       *******             *****
+
+#  ------------------------ OVERLAP TYPE: within --------------------- #
+# Range:       |==============================|
+# Annotation: |---|     |-------|           |-----| |---|
+# Selected:              *******
+
+
+class AbstractFilterRangeExpr(AbstractFilterExpr):
+    def __init__(self, value: str, type: str = "any"):
+        self.value = value
+        self.type = type
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}({self.value})"
+
+    def convert(self) -> ibis.expr.deferred.Deferred:
+        try:
+            range_seq_name, range_coords = self.value.split(":")
+            range_start, range_end = range_coords.split("-")
+        except ValueError:
+            raise ValueError(
+                "Invalid range format. Valid format is 'seq_name:start-end'"
+            )
+        start_column, end_column, seq_name_column = list(self.columns())
```

Comment:
Unlike `dict`s, `set`s don't have promised order.

I see that you're returning a list instead of a set, but also would kinda like to make this mypy-able. I don't feel super strongly about solutions, but my two ideas are:

* Have a different attribute which stores the list
* Or have all `columns()` methods return lists, and have the receiving code deal with set operations.

Both are fine, but I think the former will be easier and cause less conflicts with existing PRs